src/Installer/core-sdk-tasks/ExtractArchiveToDirectory.cs

```diff
@@ -121,8 +130,17 @@ public override bool Execute()
                                     entryName = entryName.StartsWith("./") ? entryName[2..] : entryName;
                                     if (ShouldExtractItem(entryName))
                                     {
-                                        Log.LogMessage(entryName);
                                         string destinationPath = Path.Combine(DestinationDirectory, entryName);
+                                        string destinationFileName = Path.GetFullPath(destinationPath);
+                                        string fullDestDirPath = Path.GetFullPath(DestinationDirectory + Path.DirectorySeparatorChar);
+
+                                        if (!destinationFileName.StartsWith(fullDestDirPath)) {
```

Comment:
Unless you know that both paths end with a slash, this is not a secure way of performing a path prefix check.